Tests/StreamChatUITests/AssertSnapshot.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import SnapshotTesting
+import XCTest
+
+/// Snapshot of a view controller. All variants will be tested by default. For each variant, it will take a snapshot.
+/// It is recommended to test all trait variants only for view controllers, and for small view components only test
+/// the default variant. This is to minimize the amount of pictures stored in the repo.
+///
+/// - Parameters:
+///   - vc: The view controller to be tested.
+///   - isEmbededInNavigationController: If the view controller should be embeded in a navigation controller.
+///   Useful to test the navigation bar.
+///   - variants: The variants that a snapshot will be taken. All variants should be tested for a view controller.
+///   - record: True if a new reference should be saved. False by default,
+///   so that the newly captured snapshot is compared with the current reference.
+func AssertSnapshot(
+    _ vc: UIViewController,
+    isEmbededInNavigationController: Bool = false,
+    variants: [SnapshotVariation] = SnapshotVariation.allCases,
+    record: Bool = false,
+    line: UInt = #line,
+    file: StaticString = #file,
+    function: String = #function
+) {
+    let viewController = isEmbededInNavigationController ? UINavigationController(rootViewController: vc) : vc
+    variants.forEach { variant in
+        assertSnapshot(
+            matching: viewController,
+            as: .image(traits: variant.trait),
```

Comment:
What's the default size for this? Shouldn't we specify the size explicitly?

Reply:
For view controllers, the size will be the one from the device that we are currently running on. Which means, that the CI and us when running locally, should use the same simulator device. We can improve this by running assertions on `setUp` to make sure that we are running in the correct device. We can create a `SnapshotTestCase: XCTestCase` that will make sure that when we run locally, that we are running in the correct device, because if do not, the tests will fail. 

Anyways, I will try to see if it is possible to change the width and height of the view controller to a custom one independent of the running device. But this adds another question, what if we have designed specific per iPad/iPhone ? Then, in this case, the device matters, and not the size of view controller.

Reply:
AFAIK you can provide an explicit size, you can even use something like `.image(traits: variant.trait, device: .iPhoneX)`. We need something like this. We can't rely on everybody using the same simulator for the snapshots.